### DISCUSSION TITLE
question about cache-control headers for different resources

### DISCUSSION THREAD

**Discussion by ingoclaro** [vercel/next.js#11787] [Help]
Hi, I noticed that the following cache-control headers are used for the pages:
page.js: `public, max-age=31536000, immutable`
getStaticProps json: `s-maxage=31536000, stale-while-revalidate`
getServerSideProps json: `private, no-cache, no-store, max-age=0, must-revalidate`

since getStaticProps is generated at build time and its url includes the build_id (eg `/_next/data/PskMIWblfabRNcx6K29e1/somePage.json`
why does it have those specific cache headers? seems to me that it could be cached forever, same as the page's .js file.
When building for production next even generates that json file and it is served from disc.

Seems that this is related to https://github.com/zeit/next.js/discussions/11552 but if that is the case, it should be opt in right? the default should be to return the same headers as the page.js file and if opting in to the incremental static generation, then change the headers (and perhaps getServerSideProps should be used instead since essentially the content is generated server side at that point and not at build time...)

**Comment by davecardwell** [comment]
@ingoclaro did you ever find out the reasoning behind this?

**Reply by thanhtoan1196** [reply]
@davecardwell @ingoclaro How to add `public` to the `cache-control`, I want to be cached at cdn
https://firebase.google.com/docs/hosting/manage-cache#set_cache-control
